systemtest/src/main/java/io/strimzi/systemtest/templates/specific/AdminClientTemplates.java

```diff
@@ -25,21 +33,210 @@ public class AdminClientTemplates {
 
     private AdminClientTemplates() {}
 
-    public static Deployment scramShaAdminClient(String namespaceName, String userName, String adminName, String bootstrapName, String additionalConfig) {
-        String finalAdditionalConfig = getAdminClientScramConfig(namespaceName, userName) + "\n" + additionalConfig;
+    ///////////////////////////////////////////
+    //              Tls (SSL)
+    ///////////////////////////////////////////
 
-        return defaultAdminClient(namespaceName, adminName, bootstrapName, finalAdditionalConfig).build();
+    /**
+     * Deploys an AdminClient with TLS enabled, with only essential configuration.
+     */
+    public static AdminClient deployTlsAdminClient(ResourceManager rm, String namespace, String userName, String adminName, String clusterName, String bootstrapName) {
+        return deployTlsAdminClient(rm, namespace, userName, adminName, clusterName, bootstrapName, "");
     }
 
-    public static DeploymentBuilder defaultAdminClient(String namespaceName, String adminName, String bootstrapName) {
-        return defaultAdminClient(namespaceName, adminName, bootstrapName, "");
+
+    /**
+     * Deploys an AdminClient with TLS enabled.
+     *
+     * @param rm The ResourceManager deploying admin client Pod.
+     * @param namespace The Namespace in which to deploy the AdminClient and where Kafka resides.
+     * @param userName The Kafka userName to correctly configure TLS.
+     * @param adminName The name of the AdminClient deployment.
+     * @param clusterName The name of the Kafka cluster to which the AdminClient will connect.
+     * @param bootstrapName The name of the Kafka bootstrap server to use for the initial connection.
+     * @param additionalConfig A string representing additional configuration settings for the AdminClient which will be added into configuration necessary for TLS communication.
+     * @return An instance of AdminClient configured with TLS settings.
+     */
+    public static AdminClient deployTlsAdminClient(ResourceManager rm, String namespace, String userName, String adminName, String clusterName, String bootstrapName, String additionalConfig) {
```

Comment:
Templates servers as space where you have a definition of all objects, which we use in our test suites (i.e., we have pre-defined resources that repeat many times). Here you break this contract by adding inside `Templates` a ResourceManager, which is bad. You should always use 
```java
AdminClientTemplates.tlsAdminClient(
                namespace,
                userName,
                adminName,
                clusterName,
                bootstrapName,
                additionalConfig
                ));
```
In test case and call `rm.createResourceWithWait(....` also in test case.